From 2ced15447ff4e1c254cfdf3498f37779ebd4c24c Mon Sep 17 00:00:00 2001
From: Andrea Iacovitti <aiacovitti@libero.it>
Date: Sat, 9 Jul 2011 19:51:40 +0200
Subject: [PATCH] Fix regression about input text clearButton stopping to work
 on khtml forms BUG: 246513 FIXED-IN: 4.7.0
Upstream: Yes, in master and KDE 4.7. Not in KDE 4.6.

---
 kdeui/widgets/klineedit.cpp |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/kdeui/widgets/klineedit.cpp b/kdeui/widgets/klineedit.cpp
index 7eb8bdc..09c1ebf 100644
--- a/kdeui/widgets/klineedit.cpp
+++ b/kdeui/widgets/klineedit.cpp
@@ -1095,7 +1095,7 @@ void KLineEdit::mousePressEvent( QMouseEvent* e )
     if  ( (e->button() == Qt::LeftButton ||
            e->button() == Qt::MidButton ) &&
           d->clearButton ) {
-        d->clickInClear = d->clearButton == childAt( e->pos() );
+        d->clickInClear = ( d->clearButton == childAt(e->pos()) || d->clearButton->underMouse() );
 
         if ( d->clickInClear ) {
             d->possibleTripleClick = false;
@@ -1114,7 +1114,7 @@ void KLineEdit::mousePressEvent( QMouseEvent* e )
 void KLineEdit::mouseReleaseEvent( QMouseEvent* e )
 {
     if ( d->clickInClear ) {
-        if ( d->clearButton == childAt( e->pos() ) ) {
+        if ( d->clearButton == childAt(e->pos()) || d->clearButton->underMouse() ) {
             QString newText;
             if ( e->button() == Qt::MidButton ) {
                 newText = QApplication::clipboard()->text( QClipboard::Selection );
-- 
1.7.6

